turborepo-tests/integration/tests/lockfile-aware-caching/new-package.t

```diff
@@ -0,0 +1,22 @@
+Setup
+  $ . ${TESTDIR}/../../../helpers/setup.sh
+  $ . ${TESTDIR}/setup.sh $(pwd) pnpm
+
+Add new package with an external dependency
+  $ mkdir -p apps/c
+  $ echo '{"name":"c", "dependencies": {"has-symbols": "^1.0.3"}}' > apps/c/package.json
+
+Update lockfile
+  $ pnpm i --frozen-lockfile=false > /dev/null
+
+Now build and verify that only the new package is in scope
+Note that we need --skip-infer because we've now installed a local
+turbo in this repo
+Note that we need to disable path conversion because on windows, git bash considers
+'//' to be an escape sequence translating to '/'.
+  $ MSYS_NO_PATHCONV=1 ${TURBO} --skip-infer build -F '[HEAD]' -F '!//' --dry=json | jq '.packages' 
```

Comment:
If this env var is just for the `-F '!//` flag on CLI, can you set it globally with this explanation? 

https://github.com/vercel/turbo/blob/cc565e86be3506bc80592813a192217a8cc5b0c1/turborepo-tests/helpers/setup_integration_test.sh

It's very likely that this will come up again, especially as we do more work with root packages (e.g. tracing deps through root package.json)

Reply:
I [tried it](https://github.com/vercel/turbo/actions/runs/9069220393/job/24918724152?pr=8132#step:8:941), and it appears we are relying on this behavior in other parts of our scripts. It might be possible to set it up properly, but I think that's beyond the scope of this particular PR.